File: CPP/stack.cpp
#include <bits/stdc++.h>
using namespace std;
 
class Stack {
    int top;
 
public:
    int a[100]; 
 
    Stack() 
    {
         top = -1; 
    }
    bool push(int x)
    {
        if (top >= 99) {
            cout << "Stack Overflow";
            return false;
        }
        else {
            a[++top] = x;
            cout << x << " pushed into stack\n";
            return true;
        }
    }
    int pop()
    {
        if (top < 0) {
            cout << "Stack Underflow";
            return 0;
        }
        else {
            int x = a[top--];
            return x;
        }
    }
    int peek()
    {
        if (top < 0) {
            cout << "Stack is Empty";
            return 0;
        }
        else {
            int x = a[top];
            return x;
        }
    }
    bool isEmpty()
    {
        return (top < 0);
    }
};
  
int main()
{
    class Stack s;
    s.push(10);
    s.push(20);
    s.push(30);
    s.push(40);
    s.push(50);
    cout << s.pop() << " Popped from stack\n";
    cout << s.pop() << " Popped from stack\n";
    cout<<"Elements present in stack : ";
    while(!s.isEmpty())
    {
        cout<<s.peek()<<" ";
        s.pop();
    }
 
    return 0;
}

File: PYTHON/Stack.py

from sys import maxsize
 
def createStack():
    stack = []
    return stack
 
def isEmpty(stack):
    return len(stack) == 0
 
def push(stack, item):
    stack.append(item)
    print(item + " pushed to stack ")
     
def pop(stack):
    if (isEmpty(stack)):
        return str(-maxsize -1) 
     
    return stack.pop()
 
def peek(stack):
    if (isEmpty(stack)):
        return str(-maxsize -1) 
    return stack[len(stack) - 1]
 
stack = createStack()
push(stack, str(500))
push(stack, str(120))
push(stack, str(2))
print(pop(stack) + " popped from stack")